app/serializers/search_form_serializer.rb

```diff
@@ -4,57 +4,79 @@ class SearchFormSerializer < ActiveModel::Serializer
   attributes :sort, :query
 
   def sort
-    {
-      '_script': {
-        'script': types_of_advice_sorting_expression,
-        'type': 'number',
-        'order': 'desc'
-      },
-      '_geo_distance': {
-        'advisers.location': object.coordinates.reverse,
-        'order': 'asc',
-        'unit': 'miles'
-      }
-    }.tap do |expression|
-      expression.delete(:_script) unless types_of_advice?
+    [].tap do |options|
+      if object.phone_or_online?
+        options << 'registered_name'
+      end
+
+      if types_of_advice?
+        options << {
+          _script: {
+            script: types_of_advice_sorting_expression,
+            type: 'number',
+            order: 'desc'
+          }
+        }
+      end
+
+      if object.face_to_face?
+        options << {
+          _geo_distance: {
+            'advisers.location': object.coordinates.reverse,
+            'order': 'asc',
```

Comment:
ok, agree - we just have different scopes for consistency :) yours is entire file, mine - only this hash :)